fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/SubqueryToApply.java

```diff
@@ -140,6 +140,13 @@ public List<Rule> buildRules() {
                                 .filter(s -> !(s instanceof MarkJoinSlotReference))
                                 .collect(ImmutableList.toImmutableList()), newFilter);
                     } else {
+                        if (applyPlan instanceof LogicalProject) {
```

Comment:
I think the better way to do this is using old filter's output to create a new project node on top of current result root plan

Reply:
try replace 
```
if (conjuncts.stream().flatMap(c -> c.children().stream())
                            .anyMatch(MarkJoinSlotReference.class::isInstance)) {
    return new LogicalProject<>(applyPlan.getOutput().stream()
            .filter(s -> !(s instanceof MarkJoinSlotReference))
            .collect(ImmutableList.toImmutableList()), newFilter);
} else {
    return newFilter;
}
```
with
```
return new LogicalProject<>(filter.getOutput().stream()
                            .collect(ImmutableList.toImmutableList()), newFilter);
```